### ISSUE TITLE
Practical_Machine_Learning_note

### ISSUE THREAD

**Issue by pspiskinghuayuworld** [pspiskinghuayuworld/learn_note#3]
# Practical Machine Learning
## Week One
- Components of a predict (預測的組成元素)
 question -> input data -> features (特徵) -> algorithm -> parameters -> evaluation (評測)  
- algorithm 通常是建構機器學習最不重要的部分，重要的是使用的數據類型
> the combination of some data and and aching desire for an answer does not ensure that a reasonable answer can be extracted from a given body of data.
- garbage in garbage out ，即使算法再好，數據不正確或無用的數據都不會得到好結果。
- 最重要的一步就是收集正確的數據且與嘗試回答問題相關。
- The "Best" Machine Learning Method : Interpretable (可解釋) , Simple , Accurate , Fast , Scalable(可擴展) 
- In and out sample error ，建造簡單且強大的模型，能忽略雜訊捕捉到信號
- Prediction study design
   1. Define your error rate
   2. Split data into:
       * Training, Testing, Validation(optional)驗證
   3. On the train set pick feature
       * Use cross-validation
   4. On the training set pick prediction function
       * Use cross-validation
   5. If no validation
       * Apply 1x to test set
   6. If validation
       * Apply to test set and refine
       * Apply 1x to validation
- 建立模型時一定要有完全隔離的數據庫做驗證
- 測試的 Sample 數量需夠大，才能肯定我們的預測準確率不是偶然的。
- 數據量大時：60% Training, 20% test, 20% validation
- 數據量中等時：60% Training, 40% test
- 數據量小時：採用交叉驗證，警告樣本數太小
- 預測算法 AUC 大於 0.8 才算是良好的
- K-fold cross-validation，將數據分成 K 等分後交叉驗證預測算法
- Leave on out cross-validation，只留下一組數據驗證
-
